### ISSUE TITLE
[auto-monarch] connection problem to PRACE account

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#832]
In GitLab by @ mariag on Jun 14, 2022, 09:12

Hi, 

I'm re-running an auto-monarch experiment in the PRACE account. For some reason, since this past weekend MN4 failures, autosubmit does not recognize the LOGs directory in the HPC and the process does not start. 

I've done a recovery of all jobs, and launched the experiment (both from AS01 and AS02 machines), both give the same error:

```
Autosubmit is running with v3.13.0
Checking the connection to all platforms in use
^[[32m[marenostrum_archive] Correct user privileges for host dtprace^[[0m^[[39m
^[[32m[marenostrum_archive] Connection successful to host dtprace^[[0m^[[39m
^[[32m[marenostrum4] Correct user privileges for host mn4prace^[[0m^[[39m
^[[32m[marenostrum4] Connection successful to host mn4prace^[[0m^[[39m


4517 of 11712 jobs remaining (08:43)
a4kp_20050829_000_822_ARCHIVE submitted
a4kp_20050829_000_821_CLEAN submitted


4517 of 11712 jobs remaining (08:43)
^[[32mJob a4kp_20050829_000_821_CLEAN is COMPLETED^[[0m^[[39m
Job a4kp_20050829_000_822_ARCHIVE is RUNNING
^[[38;5;214m[ERROR] Trace: None^[[0m^[[39m
^[[38;5;214m[ERROR] Command sacct -n -X --jobs  23521520,23521521 -o jobid,State in mn4prace warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused

sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused
 [eCode=6000]^[[0m^[[39m
Waiting 15 seconds before continue
```
[...]

```
4517 of 11712 jobs remaining (08:46)
^[[32mJob a4kp_20050829_000_821_CLEAN is COMPLETED^[[0m^[[39m
Job a4kp_20050829_000_822_ARCHIVE is RUNNING
^[[38;5;214m[ERROR] Trace: None^[[0m^[[39m
^[[38;5;214m[ERROR] Command sacct -n -X --jobs  23521520,23521521 -o jobid,State in mn4prace warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused
 [eCode=6000]^[[0m^[[39m
Trace A critical file couldn't be retrieved, File /gpfs/scratch/pr1enx00/pr1enx08/a4kp/LOG_a4kp does not exists
Failed to retrieve log file for job a4kp_20050829_000_821_CLEAN
Waiting 15 seconds before continue
```

Any advice? Full log can be checked here: `/esarchive/autosubmit/a4kp/runs/nohup.out`

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 14, 2022, 09:34

Hello, this seems an issue with the platform itself more than autosubmit. 

Can you execute `Command sacct -n -X --jobs 23521520,23521521` this command manually? or it also gives the slurmdbd error

**Comment by kinow** [comment]
In GitLab by @ mariag on Jun 14, 2022, 09:36

Thanks, @dbeltrankyl! I believe indeed is an issue with the platform itself. @montsecostasuros and @ yruprich seem to have the same issue... They've contacted support, let's see if they solve it quickly.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 10:28

@ mariag could you check using `squeue` if your jobs are running? According to Operations, it should only impact on `sacct`

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 10:30

`squeue` works for me at least, but I don't know if this is impacting in the jobs launched by Autosubmit or Autosubmit is cancelling them.

**Comment by kinow** [comment]
In GitLab by @ mariag on Jun 14, 2022, 10:31

The sq works also for me, but AS was not sending the jobs to the HPC... it just kept going on this loop of not "seeing" the LOG_xxx directory in the HPC.

**Comment by kinow** [comment]
In GitLab by @ mariag on Jun 14, 2022, 10:32

This: `Trace A critical file couldn't be retrieved, File /gpfs/scratch/pr1enx00/pr1enx08/a4kp/LOG_a4kp does not exists`

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 10:37

Ok, I see... my interpretation is that your `ARCHIVE` job was correctly sent and probably it finished already. I did some tests and both `sbatch` and `squeue` are working.

However, AS has no way to see the status (as `sacct` is not working).

Nevertheless, I don't understand why it cannot check the log nor the `COMPLETED` file.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 10:38

```
Trace A critical file couldn't be retrieved, File /gpfs/scratch/pr1enx00/pr1enx08/a4kp/LOG_a4kp does not exists
Failed to retrieve log file for job a4kp_20050829_000_821_CLEAN
Waiting 15 seconds before continue
```

Do you know why @dbeltrankyl ? How could we test it?

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 10:46

While we figure out why the file checks are not working the situation is the following.

Autosubmit needs `sacct` to know the status of the jobs in the workflow. It could virtually depend on the status of the `COMPLETED` files to mark the jobs as correct. This is not working well and we'll figure out why. Nevertheless, this would be like driving a car with 3 wheels and the situation is far from optimal. 

We ask you to wait until the maintenance is over, that according to Operations should take a few hours as much.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 14, 2022, 10:49

I think that autosubmit main thread is resetting the connections while the log recovery thread is trying to retrieve the log file. Maybe disabling this behaviour can work for the recovery logs ( with an impact on the performance.) 

The parameter is DISABLE_RECOVERY_THREADS = TRUE at the platform level.

However, this only works for retrieving the logs, the main issue is that:

* Autosubmit first check all jobs status. and encounters an error in the platform ( due using sacct) here instead of raise autosubmiterror it should check for complete files and if it is not available, raise an autosubmit error.
* Due to this, autosubmit submit is not reached and instead it triggers the autosubmit recovery mechanism.

Could you try with the autosubmit 3.14.0b? I think some of the modifications may help in this case

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 14, 2022, 11:06

> Could you try with the autosubmit 3.14.0b? I think some of the modifications may help in this case

I ran a dummy successfully right now

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 11:08

Thank you Dani, if someone else double-checks this with AS 3.14.0b I'll send a new message explaining it.

**Comment by kinow** [comment]
In GitLab by @ yruprich on Jun 14, 2022, 11:40

I did not have the time to check with the version 3.14 but my jobs seems to be handled by AS again.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 14, 2022, 11:43

Thanks, @ yruprich. It seems that `sacct` is operational again. I'll wait a few minutes for an official announcement and otherwise I'll send another e-mail to the list.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 11, 2022, 16:56

The issues was solved (with the end of the update) and new mechanisms were introduced into Autosubmit to be more robust against these actions.

@dbeltrankyl could you describe the changes? (on job checking and submission)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jul 21, 2022, 10:24

Basically, whenever a job is submitted and the job_id is not acquired, Autosubmit looks if the job is in sacct/ squeue by job_name and cancels it. 

This doesn't count as a failed job so, in the next iteration autosubmit tries to submit again the job. 

We could recover the job_id but I prefer to cancel them to avoid unexpected issues on running time ( that did happen to me )

**Comment by kinow** [comment]
In GitLab by @ebergas on Oct 3, 2022, 09:00

This weekend I had the same issue with sacct in mn, I wrote operations on Saturday, and now it's fixed, they told me that there was a problem with slurm database.

I am using autosubmit version 3.13, but I will definitely move to 3.14.0b for the next experiments.

**Comment by kinow** [comment]
In GitLab by @ mariag on Oct 3, 2022, 09:06

Hi, 

Same here with exps a585, a586 and a587. I moved to autosubmit 3.14.0b0 but it produced an error with the SYBCRONIZE job. Then I switched to autosubmit 3.14.0, but the sacct error persisted. 

Since this morning at 8.15 h it seems that it's working. If you wish to check the full log, see: 

`/esarchive/autosubmit/a585/runs/nohup.out`

**Comment by kinow** [comment]
In GitLab by @montsecostasuros on Oct 3, 2022, 09:17

Same problem here with a57e, a57m, a584 (bsc32 account, since PRACE has finished). I've got the message "sacct" this weekend. Now it is working after re-running the experiments.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 3, 2022, 11:32

I think this is part of a generalized problem. I created issue #879

**Comment by kinow** [comment]
In GitLab by @portegam on Oct 3, 2022, 11:54

mentioned in issue auto-ecearth3#1673

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 19, 2023, 09:15

I believe this was resolved in #879, closing this one.
